snuba/datasets/processors/spans_processor.py

```diff
@@ -0,0 +1,493 @@
+        )
+
+        processed["start_timestamp"], _ = self.__extract_timestamp(
+            event_dict["data"]["start_timestamp"],
+        )
+        if event_dict["data"]["timestamp"] - event_dict["data"]["start_timestamp"] < 0:
+            # Seems we have some negative durations in the DB
+            metrics.increment("negative_duration")
+
+        processed["end_timestamp"], _ = self.__extract_timestamp(
+            event_dict["data"]["timestamp"],
+        )
+        duration_secs = (
+            processed["end_timestamp"] - processed["start_timestamp"]
+        ).total_seconds()
+        processed["duration"] = max(int(duration_secs * 1000), 0)
+        processed["exclusive_time"] = transaction_ctx.get("exclusive_time", 0)
+
+        status = transaction_ctx.get("status", None)
+        if status:
+            int_status = SPAN_STATUS_NAME_TO_CODE.get(status, UNKNOWN_SPAN_STATUS)
+        else:
+            int_status = UNKNOWN_SPAN_STATUS
+        processed["span_status"] = int_status
+
+        processed["platform"] = _unicodify(event_dict["platform"])
+
+    @staticmethod
+    def _process_tags(
+        processed: MutableMapping[str, Any],
+        event_dict: EventDict,
+        common_span_fields: CommonSpanDict,
+    ) -> None:
+        tags: Mapping[str, Any] = _as_dict_safe(event_dict["data"].get("tags", None))
+        processed["tags.key"], processed["tags.value"] = extract_extra_tags(tags)
+
+        span_environment = _unicodify(tags.get("environment", ""))
+        release = _unicodify(tags.get("sentry:release", event_dict.get("release", "")))
+        user = _unicodify(tags.get("sentry:user", ""))
+        processed["user"] = user
+
+        common_span_fields["tags.key"] = ["environment", "release", "user"]
+        common_span_fields["tags.value"] = [span_environment, release, user]
+
+    @staticmethod
+    def _process_measurements(
+        processed: MutableMapping[str, Any],
+        event_dict: EventDict,
+    ) -> None:
+        """
+        Extracts measurements from the event_dict and writes them into the
+        measurements columns.
+        """
+        measurements = event_dict["data"].get("measurements")
+        if measurements is not None:
+            try:
+                (
+                    processed["measurements.key"],
+                    processed["measurements.value"],
+                ) = extract_nested(
+                    measurements,
+                    lambda value: float(value["value"])
+                    if (
+                        value is not None
+                        and isinstance(value.get("value"), numbers.Number)
+                    )
+                    else None,
+                )
+            except Exception:
+                logger.error(
+                    "Invalid measurements field.",
+                    extra={"measurements": measurements},
+                    exc_info=True,
+                )
+
+    @staticmethod
+    def _process_module_details(
```

Comment:
The top level span in the trace context does not have enough details to fill these in. Setting them to defaults for now.